path_static_roles_test.go

```diff
@@ -707,6 +479,47 @@ func TestListRoles(t *testing.T) {
 			t.Fatalf("expected list with %d keys, got %d", 2, len(resp.Data["keys"].([]string)))
 		}
 	})
+
+	t.Run("list roles with hierarchical role path", func(t *testing.T) {
+		b, storage := getBackend(false)
+		defer b.Cleanup(context.Background())
+
+		configureOpenLDAPMount(t, b, storage)
+
+		roles := []string{"org/secure", "org/platform/dev", "org/platform/support"}
+
+		// create all the roles
+		for _, role := range roles {
+			data := getTestStaticRoleConfig(role)
+			resp, err := createStaticRoleWithData(t, b, storage, role, data)
+			if err != nil || (resp != nil && resp.IsError()) {
+				t.Fatalf("err:%s resp:%#v\n", err, resp)
+			}
+		}
+
+		// TODO(JM): List regex should support optional trailing slash
+		// rolePaths := []string{"org", "org/", "org/platform", "org/platform/"}
+		rolePaths := []string{"org/", "org/platform/"}
+		for _, rolePath := range rolePaths {
```

Comment:
Does this test set the new `path` field? It looks like the `Data` being passed to it is `nil`, should we add a test step that also uses the new field?

Reply:
The `path` field is extracted from the request path by the framework. If it wasn't set properly, this test would fail